### ISSUE TITLE
Support for efficient sampling from `WishartMessage` and `InverseWishartMessage`

### ISSUE THREAD

**Issue by bvdmitri** [biaslab/ReactiveMP.jl#186]
Sampling from matrix variate distribution from `Distribution.jl` is quite un-optimized. This my optimised version for `InverseWishartMessage` that works 2x times faster, allocates 3x times less and gives the same result as in Distributions.jl:

```
function Distributions.rand!(rng::AbstractRNG, sampleable::ReactiveMP.InverseWishartMessage{Float64, Matrix{Float64}}, x::AbstractVector{ <: AbstractMatrix })
    
    (df, S⁻¹) = Distributions.params(sampleable)    
    S = cholinv(S⁻¹)
    
    # check 
    p = size(S, 1)
    singular = df <= p - 1
    if singular
        isinteger(df) || throw(
            ArgumentError("df of a singular Wishart distribution must be an integer (got $df)")
        )
    end
    
    A     = similar(S)
    T     = similar(S)
    l     = length(S)
    axes2 = axes(A, 2)
    
    for C in x
    
        # Wishart sample
        if singular
            r = rank(S)
            randn!(rng, view(A, :, axes2[1:r]))
            fill!(view(A, :, axes2[(r + 1):end]), zero(eltype(A)))
        else
            Distributions._wishart_genA!(rng, A, df)
        end
        Distributions.unwhiten!(S, A)
        
        M = Cholesky(A, 'L', convert(LinearAlgebra.BlasInt, 0))
        LinearAlgebra.inv!(M)
        
        unsafe_copyto!(C, 1, M.L.data, 1, l)
    end
    
    return x
end
```

**Comment by bvdmitri** [comment]
Even more optimised that does not allocate (almost):
```julia
function Distributions.rand!(rng::AbstractRNG, sampleable::ReactiveMP.InverseWishartMessage{Float64, Matrix{Float64}}, x::AbstractVector{ <: AbstractMatrix })
    
    (df, S⁻¹) = Distributions.params(sampleable)    
    S = cholinv(S⁻¹)
    L = Distributions.PDMats.chol_lower(ReactiveMP.fastcholesky(S))
    
    # check 
    p = size(S, 1)
    singular = df <= p - 1
    if singular
        isinteger(df) || throw(
            ArgumentError("df of a singular Wishart distribution must be an integer (got $df)")
        )
    end
    
    A     = similar(S)
    T     = similar(S)
    l     = length(S)
    axes2 = axes(A, 2)
    
    uv = similar(S)
    
    for C in x
    
        # Wishart sample
        if singular
            r = rank(S)
            randn!(rng, view(A, :, axes2[1:r]))
            fill!(view(A, :, axes2[(r + 1):end]), zero(eltype(A)))
        else
            Distributions._wishart_genA!(rng, A, df)
        end
        # Distributions.unwhiten!(S, A)
        lmul!(L, A) 
        
        M = Cholesky(A, 'L', convert(LinearAlgebra.BlasInt, 0))
        LinearAlgebra.inv!(M)
        
        unsafe_copyto!(C, 1, M.L.data, 1, l)
    end
    
    return x
end
```

**Comment by bvdmitri** [comment]
Fixed in #261
